zio/ndjsonio/typeparser.go

```diff
@@ -118,16 +117,16 @@ func (info *typeInfo) makeViews(data []byte) (int, error) {
 
 func appendRecordFromViews(builder *zcode.Builder, columns []zng.Column, jsonVals []jsonVal) ([]jsonVal, error) {
 
-	handleVal := func(jv jsonVal, typ zng.Type) error {
+	handleVal := func(jv jsonVal, col zng.Column) error {
 		switch jv.typ {
 		case jsonparser.Array:
 			builder.BeginContainer()
-			ztyp := zng.InnerType(typ)
+			ztyp := zng.InnerType(col.Type)
 			var iterErr error
 			callback := func(v []byte, typ jsonparser.ValueType, offset int, _ error) {
 				zv, err := parseSimpleType(v, ztyp)
 				if err != nil {
-					iterErr = err
+					iterErr = fmt.Errorf("field \"%s\" (type %s): %w", col.Name, typ, err)
```

Comment:
Since this error happens twice what do you think about making an exported error? Something like this:
```
type FieldError struct {
   Field string
   Type zng.Type
   Err error
}
```
Might be nice to start doing stuff like this so we could one day separate the warnings from the actual errors using `package` errors reflection...

Reply:
I'm all for structured errors and agree we have a lot to do there, but I'd prefer to do this when we actually get to using this warning vs error distinction somewhere (or btw also distinguishing different types of errors, such as syntax/parse error vs semantic-y errors, which is a relevant distinction for detection).